ci/main.go

```diff
@@ -0,0 +1,211 @@
+	// EXTRA ARGS FOR BUILDING (placed after the "go build")
+	// use -x for more details from a go compiler
+	EXTRA_HOST_ARGS = ""
+
+	SYSCALL_CLIENT_SIDE = "apiwasm/syscall/*.go"
+
+	REP = "g/file/" + API_VERSION + "/file.pb.go"
+)
+
+var (
+	// protobuf files
+	apiProto  string
+	testProto string
+
+	// protoc plugin
+	template    string
+	eneratorSrc string
+)
+
+func main() {
+	if err := build(context.Background()); err != nil {
+		log.Fatalf("Cannot build dagger pipeline %v\n", err)
+	}
+}
+
+func build(ctx context.Context) error {
+	fmt.Println("Building with Dagger")
+
+	// initialize Dagger client
+	// set the current(root) directory on the host as the working directory
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// get reference to the local Dockerfile
+	dockerDir := client.Host().Directory("./ci/")
+
+	// initialize image fron dockerfile
+	// mount a host directory in the container at the '/workspaces/parigot' path
+	img := client.Container().Build(dockerDir).WithDirectory("/workspaces/parigot", client.Host().Directory("."),
+		dagger.ContainerWithDirectoryOpts{Exclude: []string{".devcontainer/", "multibuild/", "build/"}}).WithWorkdir("/workspaces/parigot")
+
+	// // get version execute
+	// golang := img.WithExec([]string{"go1.20.4", "version"})
+	// version, err := golang.Stdout(ctx)
+	// if err != nil {
+	// 	return err
+	// }
+	// fmt.Println("Hello from Dagger ", version)
+
+	// // get files in the working directory
+	// contents, err := img.WithExec([]string{"ls", "/src"}).Stdout(ctx)
+	// if err != nil {
+	// 	return (err)
+	// }
+	// fmt.Println(contents)
+
+	img, err = buildProtocGenParigot(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = generateRep(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = buildFilePWasm(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	// get reference to build output directory in container
+	output := img.Directory("build")
+
+	// write contents of container build/ directory to the host
+	_, err = output.Export(ctx, "build")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildProtocGenParigot(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		TEMPLATE=$(shell find command/protoc-gen-parigot -type f -regex ".*\.tmpl")
+		GENERATOR_SRC=$(shell find command/protoc-gen-parigot -type f -regex ".*\.go")
+		build/protoc-gen-parigot: $(TEMPLATE) $(GENERATOR_SRC)
+			@rm -f $@
+			$(GO_TO_HOST) build $(EXTRA_HOST_ARGS) -o $@ github.com/iansmith/parigot/command/protoc-gen-parigot
+	*/
+
+	var err error
+	// template, err = findFilesEndWith("command/protoc-gen-parigot", ".tmpl")
+	// if err != nil {
+	// 	return img, err
+	// }
+	// eneratorSrc, err = findFilesEndWith("test", ".go")
+	// if err != nil {
+	// 	return img, err
+	// }
+	template, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.tmpl"`}).Stdout(ctx)
```

Comment:
This is a complete guess, so ignore if you don't care or it's just plain wrong:

I would expect dagger has some functions to support "searching for files recursively in a direction with a particular suffix."